CMakeLists.txt

```diff
@@ -186,7 +191,9 @@ endif()
 # We want Kokkos to be built with C++14, since that's what we're using in
 # Parthenon.
 set(Kokkos_CXX_STANDARD 14)
-if(EXISTS ${Kokkos_ROOT}/CMakeLists.txt)
+if(DEFINED EXTERNAL_KOKKOS AND EXISTS ${EXTERNAL_KOKKOS})
```

Comment:
Sorry to churn this PR more, but I don't like this approach. Here's what I say you do. Create a new option called `PARTHENON_EXTERNAL_KOKKOS`. Default it to `OFF` and default to `ON` otherwise.

```cmake
# Default to an external Kokkos package if the submodule is not populated
if(NOT EXISTS "external/Kokkos/CMakeLists.txt")
    option(PARTHENON_EXTERNAL_KOKKOS "Use an external Kokkos" ON)
else()
    option(PARTHENON_EXTERNAL_KOKKOS "Use an external Kokkos" OFF)
endif()
```

And then, later, if the external Kokkos isn't actually found, unset the cache variable so that if the user forgot to populate the git submodule, configure still works without having to re-create the build directory.

```cmake
if (PARTHENON_EXTERNAL_KOKKOS)
    find_package(Kokkos 3 REQUIRED)
    if (NOT Kokkos_FOUND)
        unset(PARTHENON_EXTERNAL_KOKKOS CACHE)
        message(FATAL_ERROR "Could not find Kokkos. Consider running `git submodule update --init` or importing a Kokkos installation into your environment)
    endif()
endif()
```

Reply:
I wasn't completely happy with this most recent approach, thanks for taking a look at this, I like your suggestion. I tinkered with your code blocks a bit to fold in the logic of using Kokkos as a subdirectory, does this look good to you? Works correctly in my tests. 
```cmake
# Default to an external Kokkos package if the submodule is not populated        
if(NOT EXISTS "external/Kokkos/CMakeLists.txt" AND NOT EXISTS "${Kokkos_ROOT}/CMakeLists.txt")
    option(PARTHENON_EXTERNAL_KOKKOS "Use an external Kokkos" ON)                
else()                                                                           
    option(PARTHENON_EXTERNAL_KOKKOS "Use an external Kokkos" OFF)               
endif()
```

```cmake
set(Kokkos_CXX_STANDARD 14)                                                      
if (PARTHENON_EXTERNAL_KOKKOS)                                                      
    find_package(Kokkos 3)                                                          
    if (NOT Kokkos_FOUND)                                                        
        unset(PARTHENON_EXTERNAL_KOKKOS CACHE)                                   
        message(FATAL_ERROR "Could not find external Kokkos. Consider importing a Kokkos installation into your environment or unsetting PARTHENON_EXTERNAL_KOKKOS")
    endif()                                                                      
else()                                                                           
  if (EXISTS ${Kokkos_ROOT}/CMakeLists.txt)                                      
    add_subdirectory(${Kokkos_ROOT} Kokkos)                                      
  elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/CMakeLists.txt)      
    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos Kokkos)         
  else()                                                                            
    message(FATAL_ERROR "Could not find Kokkos source. Consider running `git submodule update --init`, providing the path to a Kokkos source directory with Kokkos_ROOT, or setting PARTHENON_EXTERNAL_KOKKOS=ON to link to an external Kokkos installation.")
  endif()                                                                           
endif()
```

Reply:
Oops, the first block should have `"${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/CMakeLists.txt"` instead of just `"/external/Kokkos/CMakeLists.txt"

Reply:
Yeah, that's fine, but is including an external Kokkos via a subdirectory a common pattern? Haven't seen that before. It's fine to keep it, but I'm just not familiar with that pattern.